### ISSUE TITLE
Invalid ARIA Attributes

### ISSUE THREAD

**Issue by zcjess** [zcjess/Problem-Roulette#1]
Title: WCAG 4.1.2: ARIA attributes must conform to valid values (.btn-icon-dropdown)
Tags: Accessibility, WCAG 4.1.2, aria-valid-attr-value

Issue: ARIA attributes must conform to valid values (aria-valid-attr-value - https://accessibilityinsights.io/info-examples/web/aria-valid-attr-value)

Target application: Problem Roulette - https://staging.problemroulette.ai.umich.edu/StudentDashboard/

Element path: .btn-icon-dropdown

Snippet: <button aria-haspopup="true" aria-owns="dropdown_option" class="btn-icon-dropdown online-status"> <div class="status-dot active"></div> <div class="user-account"></div> <span></span></button>

Invalid ARIA attribute value: aria-owns="dropdown_option"
====

![image](https://github.com/zcjess/Problem-Roulette/assets/90011962/bafee22e-2cdc-4b07-ace7-e279a5119a4f)

**Comment by zcjess** [comment]
Title: ,WCAG 1.3.1,WCAG 4.1.2,Best Practice: ARIA role should be appropriate for the element (body)
Tags: Accessibility, , WCAG 1.3.1, WCAG 4.1.2, Best Practice, aria-allowed-role

Issue: ARIA role should be appropriate for the element (aria-allowed-role - https://dequeuniversity.com/rules/axe/4.6/aria-allowed-role?application=msftAI)

Target application: Problem Roulette - https://staging.problemroulette.ai.umich.edu/AES220/SessionManager/156386/

Element path: iframe;body

Snippet: <body tabindex="0" aria-label="Editor, id_email_message" role="textbox" aria-multiline="true" data-new-gr-c-s-check-loaded="14.1114.0" data-gr-ext-installed="" contenteditable="true" aria-readonly="false" class="cke_editable cke_editable_themed cke_contents_ltr cke_show_borders" spellcheck="true">

Environment: Chrome version 114.0.0.0
![image](https://github.com/zcjess/Problem-Roulette/assets/90011962/df92e6c8-8330-41f3-be77-42a9745223a5)

**Comment by zcjess** [comment]
Title: WCAG 4.1.2: ARIA attributes must conform to valid values (.btn-secondary[aria-haspopup="true"][aria-owns="dropdown_option"])
Tags: Accessibility, WCAG 4.1.2, aria-valid-attr-value

Issue: ARIA attributes must conform to valid values (aria-valid-attr-value - https://accessibilityinsights.io/info-examples/web/aria-valid-attr-value)

Target application: Problem Roulette - https://staging.problemroulette.ai.umich.edu/AES220/SessionHistory/156388/

Element path: .filter-dropdown > .btn-secondary[aria-haspopup="true"][aria-owns="dropdown_option"]

Snippet: <button aria-haspopup="true" aria-owns="dropdown_option" class="btn btn-secondary"><span class="glyphicon glyphicon-spacing glyphicon-filter"></span> Filter <span></span></button>

Environment: Chrome version 114.0.0.0
![image](https://github.com/zcjess/Problem-Roulette/assets/90011962/593159a2-f073-42f5-b3e4-7fb0a42e7389)
https://staging.problemroulette.ai.umich.edu/AES220/SessionHistory/156388/

**Comment by zcjess** [comment]
Title: WCAG 4.1.2: Elements must only use allowed ARIA attributes (div:nth-child(2))
Tags: Accessibility, WCAG 4.1.2, aria-allowed-attr

Issue: Elements must only use allowed ARIA attributes (aria-allowed-attr - https://accessibilityinsights.io/info-examples/web/aria-allowed-attr)

Target application: Problem Roulette - https://staging.problemroulette.ai.umich.edu/AES220/SessionHistory/156388/

Element path: div:nth-child(3) > div:nth-child(2) > div:nth-child(2)

Snippet: <div style="" aria-expanded="true">

Environment: Chrome version 114.0.0.0

====
Title: WCAG 4.1.2: Elements must only use allowed ARIA attributes (.question-response-details)
Tags: Accessibility, WCAG 4.1.2, aria-allowed-attr

Issue: Elements must only use allowed ARIA attributes (aria-allowed-attr - https://accessibilityinsights.io/info-examples/web/aria-allowed-attr)

Target application: Problem Roulette - https://staging.problemroulette.ai.umich.edu/AES220/SessionHistory/156388/

Element path: .question-response-details

Snippet: <div class="question-response-details expanded" style="" aria-expanded="true">

Environment: Chrome version 114.0.0.0

====

ARIA attribute is not allowed: aria-expanded="true"
![image](https://github.com/zcjess/Problem-Roulette/assets/90011962/529bd840-466b-4e49-8769-8eac550dc362)

**Comment by zcjess** [comment]
Title: ,WCAG 1.3.1,WCAG 4.1.2,Best Practice: ARIA role should be appropriate for the element (.pagination-list)
Tags: Accessibility, , WCAG 1.3.1, WCAG 4.1.2, Best Practice, aria-allowed-role

Issue: ARIA role should be appropriate for the element (aria-allowed-role - https://dequeuniversity.com/rules/axe/4.6/aria-allowed-role?application=msftAI)

Target application: Problem Roulette - https://staging.problemroulette.ai.umich.edu/AES220/SessionCreation/?session_type=group-study#/GroupStudy#top

Element path: .pagination-list

Snippet: <ul role="navigation" aria-label="Student Table Pages" aria-controls="user-table" class="pagination-list">

Environment: Chrome version 114.0.0.0

====
ARIA role navigation is not allowed for given element
![image](https://github.com/zcjess/Problem-Roulette/assets/90011962/cad1e445-13ae-48c8-8da0-74e10f71cb5d)
https://staging.problemroulette.ai.umich.edu/AES220/SessionCreation/?session_type=group-study#/GroupStudy#top

**Comment by zcjess** [comment]
Title: WCAG 4.1.2: Elements must only use allowed ARIA attributes (span[aria-label="Offline"])
Tags: Accessibility, WCAG 4.1.2, aria-allowed-attr

Issue: Elements must only use allowed ARIA attributes (aria-allowed-attr - https://accessibilityinsights.io/info-examples/web/aria-allowed-attr)

Target application: Problem Roulette - https://staging.problemroulette.ai.umich.edu/AES220/SessionCreation/?session_type=group-study#/GroupStudy#top

Element path: span[aria-label="Offline"]

Snippet: <span aria-label="Offline" class="status__indicator offline"></span>

aria-label attribute cannot be used on a span with no valid role attribute.

====

![image](https://github.com/zcjess/Problem-Roulette/assets/90011962/73dda905-86c1-43c3-884d-4179e3ce7473)
https://staging.problemroulette.ai.umich.edu/AES220/SessionCreation/?session_type=group-study#/GroupStudy#top
